pandas/io/common.py

```diff
@@ -67,7 +68,7 @@ def _is_url(url):
         return False
 
 
-def _expand_user(filepath_or_buffer):
+def _expand_user(filepath_or_buffer: Any) -> Any:
```

Comment:
this should be defined formally

PathOrBuffer = Union[pathlib.Path, str] (maybe need local path too)

Reply:
Yea was thinking that too. My motivation for putting Any was that these methods truly do accept anything though and could just return the caller so it's a literal interpretation of that feature, though static typing would detect potential abuses. 

Pushing up in next commit